howso/train.amlg

```diff
@@ -525,21 +525,34 @@
 							)
 					)
 
-					;create the case and output case id
-					(call !CreateCase (assoc
-						features train_features
-						feature_values
-							(if encode_features_on_train
-								(call !ConvertFromInput (assoc
-									feature_values feature_values
-									features train_features
-								))
-								;else use feature_values as-is
-								feature_values
-							)
-						session (get_value session)
-						session_training_index (+ trained_instance_count (current_index 1))
-					))
+					(seq
+						;accumulates the data mass to !dataMassChangeSinceLastAnalyze, which is either 1.0
+						; or the case weight, whichever exists.
+						(accum_to_entities (assoc
```

Comment:
this accum_to_entities winds up being called (size input_cases) number of times. this would be much more performant if you made the following change:
a) have this lambda/seq statement return the value of the if statement here (the accumulate_weight_feature value or 1.0)
b) store this entire map into a list of trained_case_weights
c) then do a single (accum_to_entities ... (apply "+" trained_case_weights))

Reply:
Didn't do this exactly the way you suggested, but instead pulled it out of the !TrainCreateCases method entirely, to similar effect.